services/sts/sts-rest/src/main/java/org/apache/cxf/sts/rest/impl/RealmSecurityConfigurationFilter.java

```diff
@@ -0,0 +1,122 @@
+
+import static java.util.Optional.ofNullable;
+import static javax.ws.rs.HttpMethod.POST;
+import static javax.ws.rs.Priorities.AUTHENTICATION;
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.WILDCARD;
+import static org.apache.cxf.jaxrs.utils.HttpUtils.getPathToMatch;
+import static org.apache.cxf.jaxrs.utils.HttpUtils.getProtocolHeader;
+import static org.apache.cxf.message.Message.HTTP_REQUEST_METHOD;
+import static org.apache.cxf.message.Message.PROTOCOL_HEADERS;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+@PreMatching
+@Priority(AUTHENTICATION)
+public class RealmSecurityConfigurationFilter implements ContainerRequestFilter {
+    public static final String REALM_NAME_PARAM = "realm";
+    private static final Logger LOG = LoggerFactory.getLogger(RealmSecurityConfigurationFilter.class);
+
+    private Map<String, Object> realmMap;
+
+    @Override
+    public void filter(ContainerRequestContext context) throws IOException {
+        final Message message = JAXRSUtils.getCurrentMessage();
+        final String realmName = getPathParams(message).getFirst(REALM_NAME_PARAM);
+
+        ofNullable(realmName)
+                .map(n -> realmMap.get(n.toUpperCase()))
+                .map(o -> (ExtRealmProperties) o)
+                .map(extRealmProperties -> extRealmProperties.getRsSecurityProperties())
+                .ifPresent(map -> {
+                    setRsSecurityProperties(message, map);
+                    JAXRSUtils.getCurrentMessage().put(REALM_NAME_PARAM.toUpperCase(), realmName);
+                });
+    }
+
+    private void setRsSecurityProperties(final Message message, final Map<String, String> properties) {
+        properties.entrySet().forEach(entry -> message.put(entry.getKey(), entry.getValue()));
+    }
+
+    private MultivaluedMap<String, String> getPathParams(final Message message) {
+        MultivaluedMap<String, String> pathParams = new MetadataMap<>();
+        try {
+            final List<ClassResourceInfo> resources = JAXRSUtils.getRootResources(message);
+            final String rawPath = getPathToMatch(message, true);
+            final Map<ClassResourceInfo, MultivaluedMap<String, String>> matchedResources = JAXRSUtils
+                .selectResourceClass(resources, rawPath, message);
+            final Map<String, List<String>> protocolHeaders = CastUtils.cast((Map<?, ?>) message.get(PROTOCOL_HEADERS));
+            final String httpMethod = getProtocolHeader(message, HTTP_REQUEST_METHOD, POST, false);
+            final List<MediaType> acceptContentTypes = JAXRSUtils.sortMediaTypes(
+                getMessageHeader(message, protocolHeaders, ACCEPT), "q");
+            final String requestContentType = getMessageHeader(message, protocolHeaders, CONTENT_TYPE);
+
+            JAXRSUtils.findTargetMethod(matchedResources, message, httpMethod, pathParams, requestContentType,
```

Comment:
I prefer my implementation. It relate to path parameter name instead it's position in path. In my opinion it is more flexible way. In case somebody need to change the path parameter name, it is easy to extend the filre class and define own name.